### PR TITLE
Improved logging and removed from error level exceptions on Handshake

### PR DESCRIPTION
**Pull request by lsebrie**

### CODE REVIEW

**Comment by rskops** [comment]
SonarQube analysis reported 16 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 6 major
* ![MINOR][MINOR] 6 minor
* ![INFO][INFO] 1 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [HandshakeHandler.java#L101](https://github.com/rsksmart/rskj/blob/b676e990298076cd5f7814aa29e754e13444362d/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L101): Method org.ethereum.net.rlpx.HandshakeHandler.decode(ChannelHandlerContext, ByteBuf, List) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![CRITICAL][CRITICAL] [HandshakeHandler.java#L180](https://github.com/rsksmart/rskj/blob/b676e990298076cd5f7814aa29e754e13444362d/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L180): Method org.ethereum.net.rlpx.HandshakeHandler.decodeHandshake(ChannelHandlerContext, ByteBuf) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![CRITICAL][CRITICAL] [HandshakeHandler.java#L242](https://github.com/rsksmart/rskj/blob/b676e990298076cd5f7814aa29e754e13444362d/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L242): Method org.ethereum.net.rlpx.HandshakeHandler.decodeHandshake(ChannelHandlerContext, ByteBuf) passes a concatenated string to SLF4J's format string [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ALO_APPENDED_STRING_IN_FORMAT_STRING)
1. ![MAJOR][MAJOR] [HandshakeHandler.java#L142](https://github.com/rsksmart/rskj/blob/b676e990298076cd5f7814aa29e754e13444362d/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L142): The Cyclomatic Complexity of this method "decodeHandshake" is 15 which is greater than 10 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [HandshakeHandler.java#L160](https://github.com/rsksmart/rskj/blob/b676e990298076cd5f7814aa29e754e13444362d/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L160): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [HandshakeHandler.java#L223](https://github.com/rsksmart/rskj/blob/b676e990298076cd5f7814aa29e754e13444362d/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L223): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [HandshakeHandler.java#L230](https://github.com/rsksmart/rskj/blob/b676e990298076cd5f7814aa29e754e13444362d/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L230): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS134)
1. ![MAJOR][MAJOR] [HandshakeHandler.java#L241](https://github.com/rsksmart/rskj/blob/b676e990298076cd5f7814aa29e754e13444362d/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L241): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [HandshakeHandler.java#L307](https://github.com/rsksmart/rskj/blob/b676e990298076cd5f7814aa29e754e13444362d/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L307): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MINOR][MINOR] [HandshakeHandler.java#L138](https://github.com/rsksmart/rskj/blob/b676e990298076cd5f7814aa29e754e13444362d/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L138): Define a constant instead of duplicating this literal "To: \t{} \tSend: \t{}" 3 times. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1192)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 13 issues
* ![MAJOR][MAJOR] 6 major
* ![MINOR][MINOR] 6 minor
* ![INFO][INFO] 1 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [HandshakeHandler.java#L142](https://github.com/rsksmart/rskj/blob/affbe9a3c6f5e7fd63c05ebd04328d78afc3e87f/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L142): The Cyclomatic Complexity of this method "decodeHandshake" is 15 which is greater than 10 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [HandshakeHandler.java#L160](https://github.com/rsksmart/rskj/blob/affbe9a3c6f5e7fd63c05ebd04328d78afc3e87f/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L160): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [HandshakeHandler.java#L223](https://github.com/rsksmart/rskj/blob/affbe9a3c6f5e7fd63c05ebd04328d78afc3e87f/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L223): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [HandshakeHandler.java#L230](https://github.com/rsksmart/rskj/blob/affbe9a3c6f5e7fd63c05ebd04328d78afc3e87f/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L230): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS134)
1. ![MAJOR][MAJOR] [HandshakeHandler.java#L241](https://github.com/rsksmart/rskj/blob/affbe9a3c6f5e7fd63c05ebd04328d78afc3e87f/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L241): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [HandshakeHandler.java#L309](https://github.com/rsksmart/rskj/blob/affbe9a3c6f5e7fd63c05ebd04328d78afc3e87f/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L309): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MINOR][MINOR] [HandshakeHandler.java#L138](https://github.com/rsksmart/rskj/blob/affbe9a3c6f5e7fd63c05ebd04328d78afc3e87f/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L138): Define a constant instead of duplicating this literal "To: \t{} \tSend: \t{}" 3 times. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [HandshakeHandler.java#L142](https://github.com/rsksmart/rskj/blob/affbe9a3c6f5e7fd63c05ebd04328d78afc3e87f/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L142): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MINOR][MINOR] [HandshakeHandler.java#L158](https://github.com/rsksmart/rskj/blob/affbe9a3c6f5e7fd63c05ebd04328d78afc3e87f/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L158): Define a constant instead of duplicating this literal "From: \t{} \tRecv: \t{}" 7 times. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [HandshakeHandler.java#L160](https://github.com/rsksmart/rskj/blob/affbe9a3c6f5e7fd63c05ebd04328d78afc3e87f/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L160): Catch Exception instead of Throwable. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1181)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 15 issues
* ![MAJOR][MAJOR] 6 major
* ![MINOR][MINOR] 6 minor
* ![INFO][INFO] 3 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [HandshakeHandler.java#L142](https://github.com/rsksmart/rskj/blob/31af796b058e9c8ace8c768a6824ba115fd8655b/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L142): The Cyclomatic Complexity of this method "decodeHandshake" is 15 which is greater than 10 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [HandshakeHandler.java#L160](https://github.com/rsksmart/rskj/blob/31af796b058e9c8ace8c768a6824ba115fd8655b/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L160): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [HandshakeHandler.java#L223](https://github.com/rsksmart/rskj/blob/31af796b058e9c8ace8c768a6824ba115fd8655b/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L223): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [HandshakeHandler.java#L230](https://github.com/rsksmart/rskj/blob/31af796b058e9c8ace8c768a6824ba115fd8655b/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L230): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS134)
1. ![MAJOR][MAJOR] [HandshakeHandler.java#L241](https://github.com/rsksmart/rskj/blob/31af796b058e9c8ace8c768a6824ba115fd8655b/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L241): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [HandshakeHandler.java#L309](https://github.com/rsksmart/rskj/blob/31af796b058e9c8ace8c768a6824ba115fd8655b/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L309): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MINOR][MINOR] [HandshakeHandler.java#L138](https://github.com/rsksmart/rskj/blob/31af796b058e9c8ace8c768a6824ba115fd8655b/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L138): Define a constant instead of duplicating this literal "To: \t{} \tSend: \t{}" 3 times. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [HandshakeHandler.java#L142](https://github.com/rsksmart/rskj/blob/31af796b058e9c8ace8c768a6824ba115fd8655b/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L142): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MINOR][MINOR] [HandshakeHandler.java#L158](https://github.com/rsksmart/rskj/blob/31af796b058e9c8ace8c768a6824ba115fd8655b/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L158): Define a constant instead of duplicating this literal "From: \t{} \tRecv: \t{}" 7 times. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [HandshakeHandler.java#L160](https://github.com/rsksmart/rskj/blob/31af796b058e9c8ace8c768a6824ba115fd8655b/rskj-core/src/main/java/org/ethereum/net/rlpx/HandshakeHandler.java#L160): Catch Exception instead of Throwable. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1181)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/main/java/org/ethereum/net/rlpx/FrameCodecHandler.java [hunk 1]
```diff
@@ -81,13 +81,11 @@ protected void encode(ChannelHandlerContext ctx, FrameCodec.Frame frame, ByteBuf
...
+            loggerNet.debug("FrameCodec failed: address {}", ctx.channel().remoteAddress(), cause);
+        } else if (cause instanceof IOException) {
+            loggerNet.info("FrameCodec failed: address {}", ctx.channel().remoteAddress(), cause);
         } else {
-            if (cause instanceof IOException) {
-                loggerNet.info(String.format("FrameCodec failed: %s", ctx.channel().remoteAddress()), cause);
-            } else {
-                loggerNet.error("FrameCodec failed: ", cause);
-            }
+            loggerNet.warn("FrameCodec failed: ", cause);
```
Comment (diega) [inline rskj-core/src/main/java/org/ethereum/net/rlpx/FrameCodecHandler.java]: I would use different messages for each situation, other than that we would  see `FrameCodec failed` on different levels and it'll be hard do identify what's the context of the logging.

Comment (lsebrie) [reply rskj-core/src/main/java/org/ethereum/net/rlpx/FrameCodecHandler.java]: There are 3 different levels, you can identify it with that. I just added some extra data.

**Pull request merged**